htsget-search/src/htsget/bam_search.rs

```diff
@@ -32,17 +32,39 @@ impl VirtualPositionExt for VirtualPosition {
   /// But when we need to translate it into a byte range, we need to make sure
   /// the reads falling inside that block are also included, which requires to know
   /// where that block ends, which is not trivial nor possible for the last block.
-  /// The simple solution goes through adding the maximum BGZF block size,
-  /// so we don't loose any read (although adding extra unneeded reads to the query results).
-  fn bytes_range_end(&self) -> u64 {
-    self.compressed() + Self::MAX_BLOCK_SIZE
+  /// The solution used here goes through reading the records starting at the compressed
+  /// part of the end virtual position(remember this will always be the start of a BGZF
+  /// block). If we read records until the compressed part of the virtual position of
+  /// the reader is different than the one we started with, we can know where
+  /// the block ends and only add to the query result the needed bytes. If
+  /// for some reason we can't read correctly the records we fall back
+  /// to adding the maximum BGZF block size
+  fn bytes_range_end(&self, reader: &mut bam::Reader<File>) -> u64 {
+    get_next_block_position(self.clone(), reader)
```

Comment:
While you are at it, can you fix up the clippy warning on clone()? Thx mate! :D

Reply:
Weird. My clippy doesn't say anything :(. I'll try updating

Reply:
Hey, maybe the Rust toolchain version in `.github` CI is too old or too new? Feel free to tweak that, with preference for newer Rust versions.

Reply:
How are you running clippy? My IDE didn't warn me and running `cargo clippy` or even `cargo clippy --all-targets --all-features -- -W clippy::pedantic` didn't get me any warnings :(

Reply:
Are you familiar with GitHub Actions? Look here: https://github.com/umccr/htsget-rs/blob/main/.github/workflows/action.yml#L27

And the GHA run and results are in: https://github.com/umccr/htsget-rs/runs/2801075528?check_suite_focus=true#step:5:62

Reply:
Okay, I don't understand exactly what happened, but now I got the warning XD